src/test/java/org/marketplace/services/AdvertisementImageManagementServiceTest.java

```diff
@@ -0,0 +1,152 @@
+package org.marketplace.services;
+
+import jakarta.persistence.EntityExistsException;
+import jakarta.persistence.EntityNotFoundException;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.marketplace.models.Advertisement;
+import org.marketplace.models.AdvertisementImage;
+import org.marketplace.repositories.AdvertisementImageManagementRepository;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import java.util.List;
+import java.util.Optional;
+
+import static org.assertj.core.api.Assertions.as;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.BDDMockito.given;
+import static org.mockito.Mockito.*;
+
+@ExtendWith(MockitoExtension.class)
+class AdvertisementImageManagementServiceTest {
+
+    @Mock
+    private AdvertisementImageManagementRepository advertisementImageManagementRepository;
+
+    @InjectMocks
+    private AdvertisementImageManagementService advertisementImageManagementService;
+
+
+    private AdvertisementImage advertisementImage;
+    private Advertisement advertisement;
+
+    @BeforeEach
+    void setUp() {
+        advertisement = new Advertisement();
+        advertisementImage = new AdvertisementImage(1L,"filePath",advertisement);
+    }
+
+    @Test
+    void testAddImage_ShouldAddImage() {
+        // given
+        given(advertisementImageManagementRepository.findById(advertisementImage.getId())).willReturn(Optional.empty());
```

Comment:
i would add real workflow here, there is no need to mock it up, we can just add stuff to h2 before each test
normally we want to use mocks as an exceptional situation